### PR TITLE
Move changes over from mother-of-all-refactors

### PR DESCRIPTION
**Pull request by norabelrose**
This is sort of a "lite" version of the much larger "mother of all refactors" PR that focuses on changes to the hidden state extraction code. It also includes the refactor of the CLI to allow for commands like `elk extract`.

### CODE REVIEW

File: elk/extraction/parser.py [hunk 1]
```diff
@@ -167,12 +112,21 @@ def get_parser():
...
+            "`--max-examples` is None; in that case the whole test set is used."
+        ),
+    )
     parser.add_argument(
         "--token-loc",
         type=str,
         default="last",
         help=(
-            "Determine which token's hidden states will be generated. Can be `first` or"
+            "Determine which token's hidden states will be extractd. Can be `first` or"
```
Comment (FabienRoger) [inline elk/extraction/parser.py:129]: typo

File: elk/training/preprocessing.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+from pathlib import Path
+import logging
+import torch
+
+
+def normalize(data: torch.Tensor):
```
Comment (FabienRoger) [inline elk/training/preprocessing.py:6]: that's not CCS' normalization
(see my PR on normalization)

Comment (FabienRoger) [inline elk/training/preprocessing.py:6]: Not the correct normalization? (see normalization PR)

File: README.md [hunk 3]
```diff
@@ -1,51 +1,37 @@
...
-
-The name of prefix can be found in `./extraction/construct_prompts.py`. This command will save hidden states to `extraction_results` and will save zero-shot accuracy to `extraction_results/extraction_results.csv`.
+To test `microsoft/deberta-v2-xxlarge-mnli` with the misleading prefix, and only the `imdb` and `amazon_polarity` datasets, while printing extra information, run:
 
-1. To train a ccs model and a logistic regression model
+1. To train a CCS model and a logistic regression model
 
 ```bash
-python train.py --model deberta-v2-xxlarge-mnli --prefix normal --dataset imdb --num-data 1000
+elk train microsoft/deberta-v2-xxlarge-mnli imdb
```
Comment (FabienRoger) [inline README.md:26]: Is this compatible with the "name" argparse? It doesn't look like it does, but I'm not familiar enough with argparse to be sure

**Review by FabienRoger — COMMENTED** [review]
I think the new

File: README.md (hunk 3 above)
Comment (norabelrose) [reply README.md:26]: Yeah this needs to be changed

File: elk/training/preprocessing.py (hunk 2 above)
Comment (norabelrose) [reply elk/training/preprocessing.py:6]: It's the normalization from the paper, but yes we should include multiple options

**Review by lauritowal — APPROVED** [review]

File: elk/extraction/extraction.py [hunk 4]
```diff
@@ -1,147 +1,151 @@
...
+    # TODO: Make this configurable or something
+    # Token used to separate the question from the answer
+    sep_token = tokenizer.sep_token or "\n"
+
+    # TODO: Maybe also make this configurable?
+    # We want to make sure the answer is never truncated
+    tokenizer.truncation_side = "left"
+
+    def tokenize(strings: list[str]):
+        return pytree_map(
```
[the remark this answers is not in the document]
Comment (norabelrose) [reply elk/extraction/extraction.py:45]: yes, it works for arbitrarily nested dictionaries, tuples, and lists https://jax.readthedocs.io/en/latest/pytrees.html

File: elk/extraction/extraction.py [hunk 5]
```diff
@@ -1,147 +1,151 @@
...
+    collator: PromptCollator,
+):
+    print(f"Using batch size: {batch_size}")
+    num_choices = len(collator.labels)
+
+    # TODO: Make this configurable or something
+    # Token used to separate the question from the answer
+    sep_token = tokenizer.sep_token or "\n"
+
+    # TODO: Maybe also make this configurable?
```
[the remark this answers is not in the document]
Comment (norabelrose) [reply elk/extraction/extraction.py:40]: Made an issue #47 and assigned it to you
